internal/handler/wallet/backup.go

```diff
@@ -25,23 +25,27 @@ func (w *walletBackupAccount) Handle(params operations.BackupAccountParams) midd
 		return resp
 	}
 
-	go handleBackup(wlt, w.prompterApp)
-
-	return operations.NewDeleteAccountNoContent()
-}
-
-func handleBackup(wlt *wallet.Wallet, prompterApp prompt.WalletPrompterInterface) {
 	promptRequest := prompt.PromptRequest{
 		Action: walletapp.Backup,
 		Msg:    fmt.Sprintf("Backup wallet %s:", wlt.Nickname),
 		Data:   nil,
 	}
-
-	_, err := prompt.WakeUpPrompt(prompterApp, promptRequest, wlt)
+	promptOutput, err := prompt.WakeUpPrompt(w.prompterApp, promptRequest, wlt)
 	if err != nil {
-		return
+		return operations.NewBackupAccountUnauthorized().WithPayload(
+			&models.Error{
+				Code:    errorCanceledAction,
+				Message: "Unable to backup wallet",
+			})
+	}
+
+	if promptOutput != nil {
+		// for private key backup, send the private key to the wails frontend
+		w.prompterApp.EmitEvent(walletapp.PromptDataEvent,
+			walletapp.EventData{Data: wlt.GetPrivKey()})
 	}
 
-	prompterApp.EmitEvent(walletapp.PromptResultEvent,
+	w.prompterApp.EmitEvent(walletapp.PromptResultEvent,
 		walletapp.EventData{Success: true, CodeMessage: utils.MsgBackupSuccess})
```

Comment:
here we have potentially 2 events emitted. to make it clearer and avoid us some traps for unit tests, should we emit an event of anther type for the private key

```go
if promptOutput != nil {
		// for private key backup, send the private key to the wails frontend
		w.prompterApp.EmitEvent(walletapp.PromptPrivateKeyEvent,
			walletapp.EventData{Data: wlt.GetPrivKey()})
	}
```

Reply:
its already the case. there is `PromptDataEvent` and `PromptResultEvent`
But it does not really help for the tests because event id is not used